qa-core/src/tests/public-api/applications/applications.spec.ts

```diff
@@ -0,0 +1,47 @@
+import TestConfiguration from "../TestConfiguration"
+import PublicAPIClient from "../PublicAPIClient"
+import generateApp from "./fixtures/generate"
+
+describe("Public API - /applications endpoints", () => {
+  const api = new PublicAPIClient()
+  const config = new TestConfiguration()
+
+  beforeAll(async () => {
+    await config.beforeAll()
+  })
+
+  afterAll(async () => {
+    await config.afterAll()
+  })
+
+  it("POST - Create a application", async () => {
+    const response = await api.post(`/applications`, {
+      body: generateApp()
+    })
```

Comment:
In this case you could used the `seedApp` function. Having utility functions for all the api operations could be very useful, something similar to: https://github.com/Budibase/budibase/blob/develop/packages/worker/src/tests/api/index.ts

For example below in the row tests we have
```ts
await api.post(`/tables/${config.testContext.table._id}/rows`, {
      body: require("./fixtures/row.json")
    })
```

Which could instead be used as 
```ts
await config.api.rows.create(tableId, row)

// with signature
const create = async (tableId: string, body: Row) => {
   return api.post(`/tables/${tableId}/rows`, { body })
}
```

This would then give us a place to hook into the public api TS definitions produced by the openapi spec build. 
The only problem is maintaining fixtures in json, if we maintained fixtures as a const we could attach the appropriate type from the openapi spec build. Here's an example scenario then: 
1) Update public api / docs
2) Types auto updated
3) Test data integrity enforced at build time

Reply:
The hardest part of this will be getting this moved into `@budibase/types` instead of server, some build tweaking needed https://github.com/Budibase/budibase/blob/master/packages/server/src/api/controllers/public/mapping/types.ts